### PR TITLE
Filtering

### PR DESCRIPTION
**Pull request by mordonez719**
## Description

Adds filters to both meal searching and exercise searching. Aside from keyword searching, users can open an additional menu for optional filters. Filters selected are converted and added to the api url for fetching applicable data.

In a future pr, a user's common/recent filters and searches will be stored to use for finding recommendations.

## Milestones
Technical Challenge #2 - Provide recommendations based on searches

## Resources
[<links to tutorials, code snippets, inspirations>
](https://react.dev/reference/react-dom/components/input)

## Test Plan
[<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>](https://www.loom.com/share/319648a4445b443dbfe04693048a9b2e?sid=ff8e010d-de95-46d5-a4aa-7711741e8c87)

### CODE REVIEW

File: front-end/src/pages/main/meals/components/MealResults.jsx [hunk 1]
```diff
@@ -20,13 +20,15 @@ function MealResults(props){
     const [apiData, fillData] = useState([]);
     const [searchQuery, setSearchQuery] = useState("");
     const [filterMap, setFilterMap] = useState({})
+    const [minCal, setMinCal] = useState(0);
+    const [maxCal, setMaxCal] = useState(3500)
```
Comment (jessiegomes) [inline front-end/src/pages/main/meals/components/MealResults.jsx:24]: 0 and 350 could be consts on their own in case the min or max ever changes. Also by giving it a variable name, the code is more readable.
